packages/common/core-interfaces/src/error.ts

```diff
@@ -0,0 +1,95 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryProperties } from "./index";
+
+/**
+ * Different error types the Container may report out to the Host
+ */
+export const ContainerErrorType = {
+	/**
+	 * Some error, most likely an exception caught by runtime and propagated to container as critical error
+	 */
+	genericError: "genericError",
+
+	/**
+	 * Throttling error from server. Server is busy and is asking not to reconnect for some time
+	 */
+	throttlingError: "throttlingError",
+
+	/**
+	 * Data loss error detected by Container / DeltaManager. Likely points to storage issue.
+	 */
+	dataCorruptionError: "dataCorruptionError",
+
+	/**
+	 * Error encountered when processing an operation. May correlate with data corruption.
+	 */
+	dataProcessingError: "dataProcessingError",
+
+	/**
+	 * Error indicating an API is being used improperly resulting in an invalid operation.
+	 */
+	usageError: "usageError",
+
+	/**
+	 * Error indicating an client session has expired. Currently this only happens when GC is allowed on a document and
+	 * aids in safely deleting unused objects.
+	 */
+	clientSessionExpiredError: "clientSessionExpiredError",
+} as const;
+export type ContainerErrorType = typeof ContainerErrorType[keyof typeof ContainerErrorType];
+
+/**
+ * Base interface for all errors and warnings at container level
```

Comment:
We should probably update the documentation to no longer be container-specific, since we're now exposing these at a lower level. Maybe just in terms of the "Fluid Framework"? Interested in others' thoughts on this, though.

Reply:
Hmm. But the name is still "ContainerErrorType". Should we also think about renaming it?

Reply:
Good point. Maybe? @anthony-murphy @markfields definitely interested in your thoughts on these moves.

Reply:
Making it just FluidErrorType, and then using the const object could be a good way to transition off the enum

Reply:
the only one that is maybe not general is the clientSession one. If i think about the future, i see some error here, that we use fluid wide, you can identify those as they have there are multiple definitions of the same type, like "usageError". those belong is core, i then see each layer's definitions, driver, container, runtime, datastore extending the common errors, and maybe parent layer with their own.